Packages/Netherlands3D/Timeline/Runtime/Scripts/ScrollRectLink.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+using System.Linq;
+
+namespace Netherlands3D.Timeline
+{
+    /// <summary>
+    /// Links scroll rects scrolling
+    /// </summary>
+    public class ScrollRectLink : MonoBehaviour
+    {
+        public ScrollRect[] scrollRects;
+
+        private ScrollRect selectedScrollRect;
+
+        [Tooltip("Fetch the Raycaster from the GameObject (the Canvas)")]
+        [SerializeField] private GraphicRaycaster m_Raycaster;
+        private PointerEventData m_PointerEventData;
+        [Tooltip("The event system of the scene")]
+        [SerializeField] private EventSystem m_EventSystem;
+
+        private void Awake()
+        {
+            // Get raycaster / event system if not assigned
+            if(m_Raycaster == null)
+            {
+                Debug.LogWarning("[ScrollRectLink] Graphic Raycaster not assigned! Finding reference in scene...");
+                m_Raycaster = FindObjectOfType<GraphicRaycaster>();
+                if(m_Raycaster == null) Debug.LogError("[ScrollRectLink] Please assign a graphic raycaster for ScrollRectLink!");
+            }
+            if(m_EventSystem == null)
+            {
+                Debug.LogWarning("[ScrollRectLink] EventSystem not assigned! Finding reference in scene...");
+                m_EventSystem = FindObjectOfType<EventSystem>();
+                if(m_EventSystem == null) Debug.LogError("[ScrollRectLink] Please assign a EventSystem for ScrollRectLink!");
+            }
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            MouseUpdate();
+        }
+
+        private void MouseUpdate()
+        {
+            //Check if the left Mouse button is clicked
+            if(Input.GetKeyDown(KeyCode.Mouse0))
```

Comment:
Ik denk dat je hier misschien beter de interface(s) kan gebruiken zoals IPointerDownHandler en IPointerUpHandler omdat het dan ook compatible is met het nieuwe InputSystem 🤔

Reply:
( Als Input Handling op Both staat gaat dit goed, maar niet wanneer alleen InputSystem wordt gebruikt )

Reply:
Heb een nieuw verbeterd script aangemaakt genaamd ScrollRectLinker.cs die alleen van de IPointerHandlers gebruik maakt